src/api/test-page-url.js

```diff
@@ -0,0 +1,48 @@
+import path from 'path';
+import OS from 'os-family';
+import { APIError } from '../errors/runtime';
+import MESSAGE from '../errors/runtime/message';
+
+const PROTOCOL_RE           = /^([\w-]+?)(?=\:)/;
+const SUPPORTED_PROTOCOL_RE = /^(https?|file):/;
+const IMPLICIT_PROTOCOL_RE  = /^\/\//;
+const ABSOLUTE_PATH_RE      = /^\/[^\/]/;
+const WIN_ABSOLUTE_PATH_RE  = /^\w:(\/|\\)/;
+const RELATIVE_PATH_RE      = /^\.\.?(\/|\\)/;
+
+
+function isAbsolutePath (url) {
+    return OS.win ? WIN_ABSOLUTE_PATH_RE.test(url) : ABSOLUTE_PATH_RE.test(url);
+}
+
+function resolveFileUrl (url, testFileName) {
+    var testFileDir = path.dirname(testFileName);
+
+    if (RELATIVE_PATH_RE.test(url))
+        url = path.join(testFileDir, url);
+
+    return 'file://' + url;
+}
+
+export function validateUrl (url, callsiteName) {
+    var protocol = url.match(PROTOCOL_RE);
+
+    if (protocol && !SUPPORTED_PROTOCOL_RE.test(url) &&
+        !(OS.win && WIN_ABSOLUTE_PATH_RE.test(url)) &&
+        url !== 'about:blank')
+        throw new APIError(callsiteName, MESSAGE.unsupportedUrlProtocol, url, protocol[0]);
+
+    return { valid: true };
```

Comment:
Why you need to return anything? Just name it `assertUrl` and let it just throw in case of invalid URL